### PR TITLE
ROX-25680: UI for Microsoft Sentinel notifier

### PR DESCRIPTION
**Pull request by vjwilson**
### Description

This change adds the following:
* A new tile for Microsoft Sentinel in the Notifiers section of the Integrations main page
* A list view for Microsoft Sentinel notifiers
* A form view, both for creating new integrations and read-only/edit view for existing integrations

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above @kcarmichael08 

### Testing and quality

- [x] the change is production ready:  the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [ ] added e2e tests.   TO COME

#### How I validated my change

Manually created, viewed, and edited some integrations (based on test values set up in Azure by @SimonBaeumer )

Tile page
Screenshot 2024-10-03 at 1 00 39 PM

List page
Screenshot 2024-10-03 at 1 04 06 PM

Read-only view
Screenshot 2024-10-03 at 1 04 25 PM

Edit view
Screenshot 2024-10-03 at 1 04 35 PM

### CODE REVIEW

**Review by SimonBaeumer — APPROVED** [review]

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:118]: Nit: the object spread will create a shallow copy of the module-level `defaultValues` here, which could result in data overwriting the defaults and being applied again on follow up visits to the form. I'd recommend we avoid this with `structuredClone` or lodash's `cloneDeep` to get a totally new object as the initial value.

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:169]: Could we add explicit types to these function parameters? Without them TypeScript is inferring them to be `any` which could cause problems in the future (and adds more places we'd have to change if we ever want to enable `noImplicitAny`...)

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:61]:
```suggestion
        name: yup.string().trim().required('A Microsoft Sentinel name is required'),
```
?

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:269]: Would this be a better accessible label?
```suggestion
                            <ToggleGroup aria-label="Authentication method selection">
```

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:308]: Nit: given the conditional rendering the second half of this condition is not necessary.

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:357]: Same here - this part of the conditional is not necessary.

File: ui/apps/platform/src/Containers/Integrations/utils/tableColumnDescriptor.ts
Comment (dvail) [inline ui/apps/platform/src/Containers/Integrations/utils/tableColumnDescriptor.ts:212]: Nit: table headers should be sentence case to match the style recommendations and other sections of the app.
```suggestion
        microsoftSentinel: [
            { accessor: 'name', Header: 'Name' },
            {
                accessor: 'microsoftSentinel.logIngestionEndpoint',
                Header: 'Log ingestion endpoint',
            },
            { accessor: 'microsoftSentinel.directoryTenantId', Header: 'Directory tenant ID' },
            { accessor: 'microsoftSentinel.applicationClientId', Header: 'Application client ID' },
        ],
```

Comment (dvail) [inline ui/apps/platform/src/Containers/Integrations/utils/tableColumnDescriptor.ts:212]: Nit: table headers should be sentence case to match the style recommendations and other sections of the app.
```suggestion
        microsoftSentinel: [
            { accessor: 'name', Header: 'Name' },
            {
                accessor: 'microsoftSentinel.logIngestionEndpoint',
                Header: 'Log ingestion endpoint',
            },
            { accessor: 'microsoftSentinel.directoryTenantId', Header: 'Directory tenant ID' },
            { accessor: 'microsoftSentinel.applicationClientId', Header: 'Application client ID' },
        ],
```

**Review by dvail — APPROVED** [review]
A handful of nits but otherwise LGTM. I don't have a ton of experience with the Integration page so up to you if you would like to wait for a second review.

**Review by dvail — APPROVED** [review]
A handful of nits but otherwise LGTM. I don't have a ton of experience with the Integration page so up to you if you would like to wait for a second review.

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:75]: Although not a blocker to merge, a possible pattern to avoid so people do not imitate in the future:
* One side of coin: Find in Files 12 results in 9 files
* Other side of coin: without another method like `required()` does `trim()` mean anything?

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:118]: Van, here is the pattern to follow:
https://github.com/stackrox/stackrox/pull/12874/files#diff-e6973e0cce8fcb46bd43624a8578812f1dca98cae2589b99edfe2f4656653872L123-L136

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:153]: Van, am available for call if we need to discuss.

The hook is second source of truth for a value that can be derived directly from Formik state.

1. Replace with assignment that is evaluated on every render:
    ```ts
    const selectedAuthMethod = … ? 'use-client-cert' : 'use-secret`;
    ```
2. Move logic from `preOnSaveHook` into `onUpdateAuthMethod` function:

    ```ts
    // clear the previous authentication method
    if (selectedAuthMethod === 'use-client-cert') {
        setFieldValue('notifier.microsoftSentinel.secret', '');
    } else {
        setFieldValue('notifier.microsoftSentinel.clientCertAuthConfig.clientCert', '');
        setFieldValue('notifier.microsoftSentinel.clientCertAuthConfig.privateKey', '');
    }
    ```

**Review by pedrottimark — CHANGES_REQUESTED** [review]

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (vjwilson) [reply ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:75]: Regarding trim without any other method, I had originally put this end as a placeholder, before writing a custom validator function, but after Simon and I discussed, we decided to keep it simple, rather than possibly over validating on initial release of the integration.

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx
Comment (vjwilson) [reply ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:153]: I actually had something similar, but Simon and I discussed and decided it was better to keep the UI state in memory. This prevents pollution the integration object with a value it doesn't store on the backend, and allows the user to explore both objects to see which one they would prefer to use.

Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx:157]: Okay, finally I get it.
1. The local state is how user chooses which credential. I was thinking originally from viewpoint after it has been set.
2. Inverse of what you experienced, I had to start this way for Quay robot accounts, but in that case, it turned out all 4 combinations (none, the one, the other, or both) of both credentials were needed in some scenario, so the form renders all of the inputs.
3. Therefore, because this is the only form that has this idiom, some comment might have been nice, for current reviewers and for future imitators within or outside of UI team of this as reference implementation.

**Review by pedrottimark — APPROVED** [review]

**Pull request merged**
